lighthouse/aggregate-service-import.md

```diff
@@ -0,0 +1,137 @@
+# Aggregate Service Import for Lighthouse
+
+## Summary
+
+The current Lighthouse implementation deviates from the MCS specification in the following ways:
+
+- ServiceExports with matching namespace/name are not combined into a single ServiceImport.
+  Each importing cluster receives multiple ServiceImports that are multiplexed at DNS lookup.
+- Conflict resolution of ServiceExports is not implemented [^1].
+
+In order to better align with the MCS specification the following functionality needs to be
+added to the Lighthouse agent:
+
+- Merge matching `ServiceExport` objects into a single `ServiceImport` object
+- Mark conflicting `ServiceExport` objects with the `ServiceExportConflict` condition
+
+There are changes required in Lighthouse's handling of `EndpointSlices` before single
+`ServiceImports` can be introduced.
+
+- The `ServiceImports` for an exported service each contain the service VIP from the exporting
+  cluster. `ServiceImports` cannot be merged like this because a `ServiceImport` cannot have
+  more than one IP. The semantics of the IP are that it should be a cluster local Service VIP,
+  if needed. The service VIPs from the exporting clusters should be stored in `EndpointSlices`,
```

Comment:
We currently use information in endpointslices for ClusterIP Services to determine if backing pods in that cluster is available or not. This is used to avoid directing queries to clusters which have no backing pods. How will this be addressed? We should withdraw that cluster's ServiceIP from EndpointSlice and add it back once at least one active pod is available.